sdk/python/feast/permissions/client/auth_client_manager_factory.py

```diff
@@ -15,8 +19,15 @@
 
 def get_auth_client_manager(auth_config: AuthConfig) -> AuthenticationClientManager:
     if auth_config.type == AuthType.OIDC.value:
-        assert isinstance(auth_config, OidcClientAuthConfig)
-        return OidcAuthClientManager(auth_config)
+        intra_communication_base64 = os.getenv("INTRA_COMMUNICATION_BASE64")
```

Comment:
we shouldn't be generating nor sending any token in intra-communication mode.
not sure why this code is needed, maybe another check is requested client-side @tmihalac ?

Reply:
We are sending the code generated token but it contains the value for intra-communication inside, the code was needed since the server containing this code yaml file was defined without username and password as a server should be defined but it is also functions as a client to the registry server and even though it is intra-communication the assert is being performed and fails
Maybe we should remove the assert instead of adding this code

Reply:
@tmihalac what's the suggested solution then?

Reply:
Keep as-is for now, thanks!

Reply:
The assert will validate the auth configuration, so let's keep this check for now until we have a better fix

Reply:
For now create another issue as mentioned https://github.com/feast-dev/feast/pull/4450#discussion_r1747385730